docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,128 @@
+in the docker daemon.  This allows you to run as the root user in a
+container while running as a non-root user in the host.
+
+Additionally, the `HostConfig` for a container now contains a parameter
+that lets you escape the private user namespace and indicate that you
+would like to [use the host's user namespace](https://github.com/docker/engine-api/blob/master/types/container/host_config.go#L310).
+
+If the Docker daemon is launched with the remap option in a Kubernetes
+cluster today and a pod author would like to run a container that uses
+the host user namespace there is not currently a way to express the
+configuration.
+
+## Assumptions
+
+1.  There is currently a bug in Docker that does not allow you to use
+`--net=host`, `--pid=host`, or `--ipc=host` when remapping is enabled.
+This is fixed in https://github.com/docker/docker/pull/25771 such that
+if you are enabling user namespace remapping you may use `--net=host`, `--pid=host`,
+and `--ipc=host` flags if `--userns=host`.
+
+## Use Cases
+
+Kubernetes cluster administrators wish to have a higher level of
+default cluster security on Docker 1.10+ by enabling user namespace
+remapping, but want to be able to continue to run cluster level actions
+for monitoring.  Assuming that the docker daemon is started with
+`--userns-remap=default`,
+here are use cases and how to achieve them:
+
+| Use Case  | Support |
+|---|---|
+| Load a kernel module  | --userns=host --cap-add SYS_ADMIN  |
+| Create a device using mknod  |  --userns=host --cap-add MKNOD |
+| Access host path as root user  | --userns=host --cap-add SYS_ADMIN  |
+| Bind to a low port on a host interface  | --userns=host, --net=host*  |
+| Monitor other processes on host looking at /proc  | --userns=host, --pid=host*  |
+| Bind to a high port on a host interface | --net=host, --userns=container_private^  |
+| Share net,pid namespaces between containers  | --net,--pid=container:id implicitly shares userns as well  |
+
+We can make a couple of generalizations from the table:
+
+1. Certain capabilities that aren't tied to any namespace such as
+CAP_MKNOD, CAP_SYS_MODULE or CAP_SYS_TIME are requested, then they
+imply userns=host.
+2. Setting any other namespace to host could imply userns=host.
+
+[*] = requires docker/docker#25771
+[^] = not supported in docker today but could be done using runc
+
+
+
+## Options
+
+1.  Provide a `HostUser` flag and require manual setting.
+
+    Pros
+    *  Quick and easy
+    *  Allows full flexibility
+
+    Cons
+    *  Requires linux specific knowledge
+
+2.  Provide a `HostUser` flag and provide auto setting.
+
+    Pros
+    *  System does the right thing when it can
+
+    Cons
+    *  Requires extra flags/config - this defaulting should only occur
+    when remapping is configured on the Docker daemon.
+    *  Does not support edge cases like using user namespaces but binding
+    to high ports on the host.
```

Comment:
Also: User might accidentally cause a container to be run in the host user ns. I'm generally weary of implicitly elevating privileges.